### DISCUSSION TITLE
react-markdown: hide frontmatter

### DISCUSSION THREAD

**Discussion by wchorski** [remarkjs/remark#1061] [Q&A] [answered]
as stated in this https://github.com/remarkjs/react-markdown/issues/464

> [ChristianMurphy](https://github.com/ChristianMurphy) "Yes, this could be done with a custom renderer for the yaml/toml frontmatter type
https://github.com/remarkjs/react-markdown#options"

The ReadMe.md has been updated since and I am lost on how to create a custom renderer. 

I use Obsidian.md to write my `.md` files. It likes frontmatter to look like this...

```
---
title: 'HOTKEYS'
date: '2022-10-10T13:07:44.675Z'
description: 'hot keys for a hot date'
---
```

When I run it through [react-markdown](https://github.com/remarkjs/react-markdown) It turns above code into a single `<h2>` tag. 

```
<h2>
title: 'HOTKEYS'
date: '2022-10-10T13:07:44.675Z'
description: 'hot keys for a hot date'
</h2>
```

Where can I learn how to create a `custom renderer` or is there a simple option I can switch?

**Comment by Murderlon** [comment, marked as answer]
Hi, if you want to support frontmatter, you can add [`remark-frontmatter`](https://github.com/remarkjs/remark-frontmatter/) as plugin:

```jsx
import remarkFrontmatter from 'remark-frontmatter'
// ...
<ReactMarkdown children={markdown} remarkPlugins={[remarkFrontmatter]} />
```
